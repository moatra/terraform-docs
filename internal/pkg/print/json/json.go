package json

import (
	"encoding/json"

	"github.com/segmentio/terraform-docs/internal/pkg/doc"
	"github.com/segmentio/terraform-docs/internal/pkg/print"
)

const (
	indent string = "  "
	prefix string = ""
)

// Print prints a document as json.
func Print(document *doc.Doc, settings *print.Settings) (string, error) {
	buffer, err := json.MarshalIndent(document, prefix, indent)
	if err != nil {
		return "", err
	}

	return string(buffer), nil
}
