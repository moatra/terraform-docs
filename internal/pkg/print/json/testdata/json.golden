{
  "variables": [
    {
      "name": "list-1",
      "type": "list",
      "default": "[\n  \"a\",\n  \"b\",\n  \"c\"\n]"
    },
    {
      "name": "list-2",
      "type": "list",
      "description": "It's list number two."
    },
    {
      "name": "list-3",
      "default": "[]"
    },
    {
      "name": "long_type",
      "type": "object({\n    name = string,\n    foo  = object({ foo = string, bar = string }),\n    bar  = object({ foo = string, bar = string }),\n    fizz = list(string),\n    buzz = list(string)\n  })",
      "description": "This description is itself markdown.\n\nIt spans over multiple lines.\n",
      "default": "{\n  \"bar\": {\n    \"bar\": \"bar\",\n    \"foo\": \"bar\"\n  },\n  \"buzz\": [\n    \"fizz\",\n    \"buzz\"\n  ],\n  \"fizz\": [],\n  \"foo\": {\n    \"bar\": \"foo\",\n    \"foo\": \"foo\"\n  },\n  \"name\": \"hello\"\n}"
    },
    {
      "name": "map-1",
      "type": "map",
      "default": "{\n  \"a\": 1,\n  \"b\": 2,\n  \"c\": 3\n}"
    },
    {
      "name": "map-2",
      "type": "map",
      "description": "It's map number two."
    },
    {
      "name": "map-3",
      "default": "{}"
    },
    {
      "name": "string-1",
      "default": "\"bar\""
    },
    {
      "name": "string-2",
      "type": "string",
      "description": "It's string number two."
    },
    {
      "name": "string-3",
      "default": "\"\""
    },
    {
      "name": "unquoted"
    },
    {
      "name": "variable_with_underscores"
    }
  ],
  "outputs": [
    {
      "name": "output-0.12",
      "description": "terraform 0.12 only"
    },
    {
      "name": "output-1"
    },
    {
      "name": "output-2",
      "description": "It's output number two."
    },
    {
      "name": "unquoted",
      "description": "It's unquoted output."
    }
  ],
  "providers": [
    {
      "name": "aws",
      "version": "\u003e= 2.15.0"
    },
    {
      "name": "aws",
      "alias": "ident",
      "version": "\u003e= 2.15.0"
    }
  ]
}
